x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/rules/10_query_ruleset_put.yml

```
@@ -160,3 +170,51 @@ teardown:
                 - 'id2'
 
   - match: { error.type: 'security_exception' }
+
+---
+"Test numeric comparison rule validation":
+  - skip:
+      features: headers
+
+  - requires:
+      cluster_features: "query_rules.numeric_validation"
+      reason: "Query rules enforce numeric validation starting with 8.19"
+
+  - do:
+      catch: /\[query_ruleset\] failed to parse field \[rules\]/
```

Can we please catch the string pertaining to the error we're throwing here, to ensure that we're actually throwing the error that we think we are throwing? 

When you run a PUT command with an invalid ruleset, the response is: 
```
{
  "error": {
    "root_cause": [
      {
        "type": "x_content_parse_exception",
        "reason": "Failed to build [query_rule] after last required field arrived"
      }
    ],
    "type": "x_content_parse_exception",
    "reason": "Failed to build [query_rule] after last required field arrived",
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "Input [bar] is not valid for CriteriaType [lte]"
    }
  },
  "status": 400
}
```

You can search on any part of that response, not just the top level parse error. You can see the reason at the bottom is an IllegalArgumentException with a more informative message: `Input [bar] is not valid for CriteriaType [lte]`

```suggestion
      catch: /Input \[not_a_number\] is not valid for CriteriaType \[lte\]/
```